exercises/IaC/ansible/inventory/.templates/README.md.erb

```diff
@@ -0,0 +1,226 @@
+# Inventory
+
+## Introduction
+Ansible requires a list of hosts to execute playbooks and ad-hoc commands against.  Ansible refers
+to this target host list as an inventory.  This exercise will outline some basic inventory concepts -
+inventory formats, grouping, setting and using variables.
+
+## Exercise Learning Objectives
+
+  - Provide an overview of static inventories
+  - Introduce hostgroups and variables
+  - Create a static inventory with hostgroups and variables
+
+## Introduction to Static Inventories
+
+### Formats
+
+Ansible supports a number of inventory file formats:
+
+  - INI style static files
+  - JSON/YAML static files
+  - Any executable that outputs valid JSON/YAML
+  - Directories; Ansible will read all inventory files in the specified directory
+  - Python plugins
+ 
+You can find mode information on supported inventory formats in the official 
+[Ansible documentation](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html).
+
+### Basic Inventory
+
+Ansible commands will look for the default inventory file in `/etc/ansible/hosts`.  The inventory can be overridden
+on the command line using the `-i INVENTORY` option or via the `ansible.cfg` config file.
+
+The default inventory file can be a simple list of hostnames:
+
+```
+server1
+server2
+server3
+```
+
+### Inventory Groups
+
+As well as a simple list of hosts, inventories allow hosts to be grouped:
+
+```
+[webservers]
+server1
+
+[dbservers]
+server2
+server3
+
+[dc1hosts]
+server2
+
+[dc2hosts]
+server1
+server3
+```
+
+Host can be defined in more than one group.
+
+Groups can be used to limit the scope of playbooks and ad-hoc commands, we can target groups using the
+`hosts:` parameter in our playbooks.  For example, take the following playbook as an example:
+
+```YAML
+---
+- hosts: dc1hosts
+  tasks:
+    - name: Run command (only on hosts in DC1)
+      command: /bin/somecommand
+```
+
+Tasks in the play will only run on hosts in the `dc1hosts` group.
+
+Restrictions can also be placed via the `-l LIMIT` option on the command line:
+
+```
+ansible -m ping all -l dc2hosts
+```
+
+In this case, the `all` built-in group is initially specified but is subsequently limited to hosts only in the
+`dc2hosts` group.
+
+### Inventory Variables
+
+Variables can be defined in the inventory file and made available to the Ansible run, variables can be set,
+per host, within the inventory file:
+
+```
+[webservers]
+server1   dns_server=1.1.1.1
+
+[dbservers]
+server2   dns_server=2.2.2.2
+server3   dns_server=3.3.3.3
```

Comment:
Would be nice to show an example of multiple variables being used within an inventor. Are they comma separated, space separated, semicolon separated? ... If one of these examples was showing multiple vars being defined then I'd know.